documentation/api/io.strimzi.api.kafka.model.listener.arraylistener.GenericKafkaListener.adoc

```diff
@@ -0,0 +1,323 @@
+
+By specifying a unique name and port for each listener,
+you can configure multiple listeners.
+
+=== `type`
+
+The type is set as `internal`,
+or for external listeners, as `route`, `loadbalancer`, `nodeport` or `ingress`.
+
+internal:: You can configure internal listeners with or without encryption using the `tls` property.
++
+.Example `internal` listener configuration
+[source,yaml,subs="+attributes"]
+----
+#...
+spec:
+  kafka:
+    #...
+    listeners:
+      #...
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: tls
+    #...
+----
+route:: Configures an external listener to expose Kafka using OpenShift `Routes` and the HAProxy router.
++
+A dedicated `Route` is created for every Kafka broker pod.
+An additional `Route` is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Routes` to connect to Kafka on port 443.
++
+.Example `route` listener configuration
+[source,yaml,subs="+attributes"]
+----
+#...
+spec:
+  kafka:
+    #...
+    listeners:
+      #...
+      - name: external1
+        port: 9094
+        type: route
+        tls: true
+    #...
+----
+ingress:: Configures an external listener to expose Kafka using Kubernetes `Ingress` and the {NginxIngressController}.
++
+A dedicated `Ingress` resource is created for every Kafka broker pod.
+An additional `Ingress` resource is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Ingress` resources to connect to Kafka on port 443.
++
+You must specify the hostnames used by the bootstrap and per-broker services
+using xref:type-GenericKafkaListenerConfigurationBootstrap-reference[`GenericKafkaListenerConfigurationBootstrap`]
+and xref:type-GenericKafkaListenerConfigurationBroker-reference[`GenericKafkaListenerConfigurationBroker`] properties.
++
+.Example `ingress` listener configuration
+[source,yaml,subs="+attributes"]
+----
+#...
+spec:
+  kafka:
+    #...
+    listeners:
+      #...
+      - name: external2
+        port: 9095
+        type: ingress
+        tls: false
+        authentication:
+          type: tls
+        configuration:
+          bootstrap:
+            host: bootstrap.myingress.com
+          brokers:
+          - broker: 0
+            host: broker-0.myingress.com
+          - broker: 1
+            host: broker-1.myingress.com
+          - broker: 2
+            host: broker-2.myingress.com
+  #...
+----
++
+NOTE: External listeners using `Ingress` are currently only tested with the {NginxIngressController}.
+
+
+loadbalancer:: Configures an external listener to expose Kafka `Loadbalancer` type `Services`.
++
+A new loadbalancer service is created for every Kafka broker pod.
+An additional loadbalancer is created to serve as a Kafka _bootstrap_ address.
+Loadbalancers listen to connections on port 9094.
```

Comment:
I was confused initially. It's not clear that you are referring to the example, are you? The sentence sounds like that loadbalancer always listens on port 9094 :-)

Reply:
It does.
I've changed to:

> Loadbalancers listen to the specified port number, which is port `9094` in the following example.